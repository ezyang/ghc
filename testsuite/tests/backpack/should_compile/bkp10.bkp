unit p where
    signature H where
        data S
    module A where
        import H
        data T = T S

unit q where
    include p (A as A2) requires (H as H2)
    module B where
        import A2
        import H2
        t = T :: S -> T
