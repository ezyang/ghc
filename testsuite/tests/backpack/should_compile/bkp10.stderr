[1 of 2] Processing p
/--- Shape for p
provides: A -> p(hole:H):A
               T{T, T}
requires: H -> p(hole:H):H
               S{S}
signature hole:H
module p(hole:H):A
\---
  [1 of 2] Compiling H[abstract sig]  ( p/H.hsig, nothing )
  [2 of 2] Compiling A                ( p/A.hs, nothing )
[2 of 2] Processing q
/--- Shape for q
provides: A2 -> p(H -> hole:H2):A
                T{T, T}
          B -> q(hole:H2):B
               t
requires: H2 -> p(H -> hole:H2):H
                S{S}
include p(H -> hole:H2) [(A2, p(H -> hole:H2):A)] requires [(H2,
                                                             [p(H -> hole:H2):H])]
module q(hole:H2):B
\---
  [1 of 2] Including p
  [2 of 2] Compiling B                ( q/B.hs, nothing )
